mapie/tests/test_regression.py

```diff
@@ -252,6 +259,45 @@ def test_predict_output_shape(
     assert y_pis.shape == (X.shape[0], 2, n_alpha)
 
 
+@pytest.mark.parametrize("delta", [0.6, 0.8])
+@pytest.mark.parametrize("n_calib", [10 + i for i in range(11)] + [50, 100])
+def test_coverage_validity(delta: float, n_calib: int) -> None:
+    """
+    Test that the prefit method provides valid coverage
+    for different calibration data sizes and coverage targets.
+    """
+    n_split, n_train, n_test = 100, 100, 1000
+    n_all = n_train + n_calib + n_test
+    X, y = make_regression(n_all, random_state=random_state)
+
+    X_train, X_cal_test, y_train, y_cal_test = \
+        train_test_split(X, y, train_size=n_train, random_state=random_state)
+
+    model = LinearRegression()
+    model.fit(X_train, y_train)
+
+    cov_list = []
+    for _ in range(n_split):
+        mapie_reg = MapieRegressor(estimator=model, method="base", cv="prefit")
+        X_cal, X_test, y_cal, y_test = \
+            train_test_split(X_cal_test, y_cal_test, test_size=n_test)
+        mapie_reg.fit(X_cal, y_cal)
+        _, y_pis = mapie_reg.predict(X_test, alpha=1-delta)
+        coverage = \
+            regression_coverage_score(y_test,  y_pis[:, 0, 0], y_pis[:, 1, 0])
+        cov_list.append(coverage)
+
+    # Here we are testing whether the average coverage is statistically
+    # less than the target coverage.
+    from scipy.stats import ttest_1samp
+    mean_low, mean_up = delta, delta + 1/(n_calib+1)
+    _, pval_low = ttest_1samp(cov_list, popmean=mean_low, alternative='less')
+    _, pval_up = ttest_1samp(cov_list, popmean=mean_up, alternative='greater')
+
+    np.testing.assert_array_less(0.01, pval_low)
```

Comment:
What is the probability of this breaking?

Reply:
I propose to run a FWER control procedure (Bonferroni correction) with p_fwer = 1% (see next update).